contributions/views.py

```diff
@@ -73,6 +74,59 @@ def perform_create(self, serializer):
             action='CONTRIBUTION',
             details=f"Manual contribution of KES {serializer.validated_data['amount']:.2f}"
         )
+
+
+    def partial_update(self, request, *args, **kwargs):
+        try:
+            contribution = self.get_object()
+        except Contribution.DoesNotExist:
+            return Response({"error": "Contribution not found."}, status=status.HTTP_404_NOT_FOUND)
+        permission_checker = IsChamaMember()
+        if not permission_checker.has_object_permission(request, self, contribution):
+            raise PermissionDenied("You do not have permission to update this contribution. Only admins and owners of this contribution can update contributions.", status=status.HTTP_403_FORBIDDEN)
```

Comment:
Consider overriding `get_permissions` class in this  for a cleaner implementation and to avoid repeating raise `PermissionDenied`, same case with delete.